nav2_shim_controller/src/shim_controller.cpp

```diff
@@ -0,0 +1,397 @@
+  declare_parameter_if_not_declared(
+    node, plugin_name_ + ".max_linear_vel", rclcpp::ParameterValue(0.26));
+  declare_parameter_if_not_declared(
+    node, plugin_name_ + ".max_angle_threshold", rclcpp::ParameterValue(0.785));
+  declare_parameter_if_not_declared(
+    node, plugin_name_ + ".lookahead_dist", rclcpp::ParameterValue(0.4));
+  declare_parameter_if_not_declared(
+    node, plugin_name_ + ".default_plugin", rclcpp::PARAMETER_STRING);
+  declare_parameter_if_not_declared(
+    node, plugin_name_ + ".use_rotate_to_heading", rclcpp::ParameterValue(true));
+  declare_parameter_if_not_declared(
+    node, plugin_name_ + ".use_rotate_to_path", rclcpp::ParameterValue(true));
+  declare_parameter_if_not_declared(
+    node, plugin_name_ + ".use_dynamic_threshold", rclcpp::ParameterValue(true));
+
+  node->get_parameter(plugin_name_ + ".transform_tolerance", transform_tolerance);
+  node->get_parameter(plugin_name_ + ".max_angular_accel", max_angular_accel_);
+  node->get_parameter(plugin_name_ + ".max_angular_vel", max_angular_vel_);
+  node->get_parameter(plugin_name_ + ".max_linear_vel", max_linear_vel_);
+  desired_linear_vel_ = max_linear_vel_;
+  node->get_parameter(plugin_name_ + ".max_angle_threshold", max_angle_threshold_);
+  node->get_parameter(plugin_name_ + ".lookahead_dist", lookahead_dist_);
+  node->get_parameter(plugin_name_ + ".use_rotate_to_heading", use_rotate_to_heading_);
+  node->get_parameter(plugin_name_ + ".use_rotate_to_path", use_rotate_to_path_);
+  node->get_parameter(plugin_name_ + ".use_dynamic_threshold", use_dynamic_threshold_);
+  node->get_parameter("controller_frequency", control_frequency);
+
+  transform_tolerance_ = tf2::durationFromSec(transform_tolerance);
+  control_duration_ = 1.0 / control_frequency;
+
+  if (node->get_parameter(plugin_name_ + ".default_plugin", default_plugin_name)) {
+    std::string default_plugin_type = nav2_util::get_plugin_type_param(node, default_plugin_name);
+    default_plugin = plugin_loader_.createSharedInstance(default_plugin_type);
+    RCLCPP_INFO(
+      logger_, "Initializing default plugin \"%s\" with type \"%s\"",
+      default_plugin_name.c_str(), default_plugin_type.c_str());
+
+    try {
+      default_plugin->configure(node, default_plugin_name, tf_, costmap_ros_);
+    } catch (const std::exception & e) {
+      RCLCPP_ERROR(logger_, "Couldn't initialize default plugin! %s", e.what());
+      throw nav2_core::PlannerException("Unable to load default plugin!");
+    }
+    RCLCPP_INFO(logger_, "Initialized default plugin \"%s\"", default_plugin_name.c_str());
+  } else {
+    RCLCPP_ERROR(
+      node->get_logger(), "Can not get 'default_plugin' param value for %s", plugin_name_.c_str());
+  }
+}
+
+void ShimController::cleanup()
+{
+  RCLCPP_INFO(
+    logger_,
+    "Cleaning up controller: %s of type"
+    " shim_controller::ShimController",
+    plugin_name_.c_str());
+
+  if (default_plugin) {default_plugin->cleanup();}
```

Comment:
Expand onto multiple lines (and for all cases of this below) to match styling of the rest of the program.

Reply:
Also need to `reset()` the pointer, I believe in cleanup.